src/lib/bootloader/device_map.rb

```diff
@@ -92,9 +94,7 @@ def reduce_to_bios_limit
         return false
       end
 
-      grub_devices = @model.grub_devices
-      grub_devices.select! { |d| d.start_with?("hd") }
-      grub_devices.sort_by(&:to_i)
```

Comment:
@mvidner yeah, it was the reason of the previous comment but also it is not using "!" so grub_devices is not affected after sort_by and you are not assigned to any variable.